packages/libs/react-ui/src/components/Tooltip/Tooltip.tsx

```diff
@@ -1,21 +1,52 @@
-import React, { forwardRef } from 'react';
-import { arrowVariants, container } from './Tooltip.css';
-
-export interface ITooltipProps {
-  placement?: 'top' | 'bottom' | 'left' | 'right';
-  children: React.ReactNode;
+import { Box } from '@components/Layout';
+import { mergeRefs } from '@react-aria/utils';
+import type { FC, ReactElement, ReactNode } from 'react';
+import React, { cloneElement, useRef } from 'react';
+import { useTooltip, useTooltipTrigger } from 'react-aria';
+import type { TooltipTriggerProps } from 'react-stately';
+import { useTooltipTriggerState } from 'react-stately';
+import { tooltipPositionVariants } from './Tooltip.css';
+export interface ITooltipProps extends Omit<TooltipTriggerProps, 'trigger'> {
+  children: ReactElement;
+  content: ReactNode;
+  position?: keyof typeof tooltipPositionVariants;
 }
 
-export const Tooltip: React.ForwardRefExoticComponent<
-  Omit<ITooltipProps, 'ref'> & React.RefAttributes<HTMLDivElement>
-  // eslint-disable-next-line react/display-name
-> = forwardRef<HTMLDivElement, ITooltipProps>(
-  ({ children, placement = 'right' }, ref) => {
-    return (
-      <div className={container} ref={ref} data-placement={placement}>
-        <div className={arrowVariants[placement]} />
-        {children}
-      </div>
-    );
-  },
-);
+export const Tooltip: FC<ITooltipProps> = ({
+  children,
+  content,
+  position = 'right',
+  ...props
+}) => {
+  const config = {
```

Comment:
Should we give this a type?

Reply:
It is implicitly typed since the props are already typed. I see some of them are missing from storybook though so I will add them 👍